test/cpp/jit/test_memory_planning.cpp

```diff
@@ -0,0 +1,48 @@
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+#include <ATen/core/interned_strings.h>
+#include <ATen/core/ivalue.h>
+#include <torch/csrc/jit/runtime/profiling_record.h>
+#include <torch/csrc/jit/runtime/profiling_graph_executor_impl.h>
+#include <torch/csrc/jit/passes/memory_planning.h>
+#include <torch/csrc/jit/passes/tensorexpr_fuser.h>
+
+#include <test/cpp/jit/test_utils.h>
+
+namespace torch {
+namespace jit {
+
+TEST(MemoryPlannerTest, Basic) {
+  // setup inputs
+  constexpr int batch_size = 4;
+  constexpr int input_size = 256;
+  int hidden_size = 2 * input_size;
+  auto input = at::randn({batch_size, input_size}, at::kCPU);
+  auto hx = at::randn({batch_size, hidden_size}, at::kCPU);
+  auto cx = at::randn({batch_size, hidden_size}, at::kCPU);
+  auto w_ih = at::randn({4 * hidden_size, input_size}, at::kCPU).t();
+  auto w_hh = at::randn({4 * hidden_size, hidden_size}, at::kCPU).t();
+  auto g = build_lstm();
+  auto stack = createStack({input, hx, cx, w_ih, w_hh});
+
+  // run once to type info
+  auto pr = jit::ProfilingRecord::instrumentGraph(g);
```

Comment:
are you using Profiler for prototyping or we would be also using it in the final version? I'm asking because if we will be using it in the final version we will need to figure out how we will guard if shapes/strides/device change?

Reply:
I started out using Profiler for prototyping before I wrote this https://github.com/pytorch/pytorch/pull/62377. Will transition soon.

Reply:
Yeah, this makes sense!